packages/sync-actions/src/utils/create-build-actions.js

```diff
@@ -16,6 +97,10 @@ export default function createBuildActions(differ, doMapActions, onBeforeDiff) {
       onBeforeDiff
     )
 
+    if (processedNow.variants && processedBefore.variants)
+      // run updateMissingPriceIds function to mutate processedNow data
+      updateMissingPriceIds(processedNow.variants, processedBefore.variants)
```

Comment:
This could be then

```js
processedNow.variants = updateMissingPriceIds(processedNow.variants, processedBefore.variants)
```

where you'd maybe need an eslint ignore next-line if it barks due to mutation.